tracer/test/Datadog.Trace.Tests/Util/UriHelpersTests.cs

```diff
@@ -53,7 +53,29 @@ public void GetCleanUriPath_ShouldRemoveIdsFromPaths(string url, string expected
         [InlineData("ftp://example.org/controller/action/2022.png", "/controller/action/?.png")]
         public void GetCleanUriPath_ByUri_ShouldExtractThePathAndRemoveIds(string url, string expected)
         {
-            Assert.Equal(expected, Trace.Util.UriHelpers.GetCleanUriPath(new Uri(url)));
+            Assert.Equal(expected, Trace.Util.UriHelpers.GetCleanUriPath(new Uri(url), virtualPathToRemove: null));
+        }
+
+        [Theory]
+        [InlineData("http://localhost:5040", "", "/")]
+        [InlineData("http://localhost:5040", "/", "/")]
+        [InlineData("http://localhost:5040/", "", "/")]
+        [InlineData("http://localhost:5040/", "/", "/")]
+        [InlineData("http://localhost:5040/controller/", "", "/controller/")]
+        [InlineData("http://localhost:5040/controller/", "/", "/controller/")]
+        [InlineData("http://localhost:5040/controller/", "/Some-value", "/controller/")]
+        [InlineData("http://localhost:5040/Some-value/controller/", "", "/Some-value/controller/")]
+        [InlineData("http://localhost:5040/Some-value/controller/", "/", "/Some-value/controller/")]
+        [InlineData("http://localhost:5040/Some-value/controller/", "/Some-value", "/controller/")]
```

Comment:
Is it possible for the prefix to be `/Some-value/controller/`, in other words nested twice?

Reply:
The final `[InlineData]` shows an example of that. The upshot is that we should only remove it _once_:

```csharp
[InlineData("https://example.org/sup/sup/controller/action/2022", "/sup", "/sup/controller/action/?")]

```

Reply:
oh I see, good question 🤔 Yes, I think it is (you can have sub apps of a sub app)
![image](https://user-images.githubusercontent.com/18755388/171345129-5ccba6bf-60fe-4324-bea3-c65a2c2f30de.png)

I'll add some unit tests for that scenario